app/models/config.rb

```diff
@@ -70,4 +76,40 @@ def self.start_day
     start ||= "monday"
     start.downcase.to_sym
   end
+
+
+  def validate_urls
+    url = options.try :last
+
+    # empty url is allowed
+    return if url.blank?
+
+    if not url =~ /\A#{URI::regexp(['https'])}\z/
+      errors.add :base, "\"#{url}\" is not a valid URL for #{config_key.humanize}."
+    end
+  end
+
+
+  def clean_urls
+    url = options.try :last
+
+    # don't try to clean empty url
+    return if url.blank?
+
+    # don't have to do anything, already https
+    return if url.start_with? 'https://'
+
+    # convert http or // to https://
+    if url.start_with? /(http:)?\/\//
```

Comment:
Would this stuff make more sense in `UriService`? This is quite a bit of very particular uri handling, wonder if it makes sense to extract it into our service class.